pinot-core/src/main/java/org/apache/pinot/core/util/SchemaUtils.java

```diff
@@ -39,65 +48,134 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(SchemaUtils.class);
 
+  private static final SqlAbstractParserImpl.Metadata SQL_PARSER_METADATA = SqlParser.create("",
```

Comment:
I would not take a sql dependency here. I think it is cleaner to separate the query language used from Pinot schema. No matter which query language we use, there will be some reserved words.
I would have a sql init call into schema utils init to set the reserved words here.